### PR TITLE
coord,dataflow: expose librdkafka stats more directly

### LINKED ISSUE #7172: Consider alternative design of system tables for librdkafka statistics
We currently expose two system tables with librdkafka metrics:

```
show columns from mz_kafka_broker_rtt;
     name      | nullable |  type  
---------------+----------+--------
 consumer_name | f        | text
 source_id     | f        | text
 dataflow_id   | f        | bigint
 broker_name   | f        | text
 min           | f        | bigint
 max           | f        | bigint
 avg           | f        | bigint
 sum           | f        | bigint
 cnt           | f        | bigint
 stddev        | f        | bigint
 p50           | f        | bigint
 p75           | f        | bigint
 p90           | f        | bigint
 p95           | f        | bigint
 p99           | f        | bigint
 p99_99        | f        | bigint
(16 rows)

materialize=> show columns from mz_kafka_consumer_partitions;
     name      | nullable |  type  
---------------+----------+--------
 consumer_name | f        | text
 source_id     | f        | text
 dataflow_id   | f        | bigint
 partition_id  | f        | text
 rx_msgs       | f        | bigint
 rx_bytes      | f        | bigint
 tx_msgs       | f        | bigint
 tx_bytes      | f        | bigint
 lo_offset     | f        | bigint
 hi_offset     | f        | bigint
 ls_offset     | f        | bigint
 app_offset    | f        | bigint
 consumer_lag  | f        | bigint
(13 rows)
```

These are basically rote flattenings of the statistics structs provided by librdkafka: https://github.com/edenhill/librdkafka/blob/master/STATISTICS.md

There are several things worth considering:

  1. The column names don't follow our convention for system tables (e.g. `cnt` instead of `count`).
  1. We're very tied to librdkafka's statistics shape. If we ever replace librdkafka as our Kafka library, we'll have to rework these tables, or at least go to some trouble to mash the new library's statistics format into this format.
  1. We go to a _lot_ of trouble to unpack all of these values and ship them around our logging infrastructure.

An alternative for consideration: what if we just shipped the raw statistics reported by librdkafka as a big bag of `jsonb`? That would increase our coupling to the librdkafka format only marginally, and reduce the maintenance burden of these tables to near zero. New librdkafka releases that add statistics would be immediately accessible upon upgrade.

cc @elindsey for thoughts

### PR DESCRIPTION
**Pull request by benesch**
Maintaining the mz_kafka_consumer_partitions and mz_kafka_broker_rtt
tables has proven to be a maintenance burden because the statistics from
the underlying librdkafka library are not consistent from version to
version. Rather than tightly coupling the shape of our system tables to
the shape of the librdkafka statistics, expose just one system table
named `mz_kafka_source_statistics`, which exposes the JSON payload from
librdkafka directly.

This does make manipulating these statistics a bit harder for our users,
but:

  * Only our most sophisticated users access these statistics.

  * The additional difficulty of manipulating the statistics is probably
    outweighed by the fact that *all* the librdkafka statistics are
    available, not just the ones we've gone to the effort of extracting.

Fix #7172.

### Motivation

   * This PR refactors existing code.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR adds a release note for any
  [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

**Review by elindsey — APPROVED** [review]
yes please 🙏

**Comment by umanwizard** [comment]
No strong opinion from me on this

File: src/dataflow-types/src/logging.rs [hunk 1]
```diff
@@ -269,13 +240,9 @@ impl LogVariant {
...
             LogVariant::Materialized(MaterializedLog::FrontierCurrent) => vec![],
-            LogVariant::Materialized(MaterializedLog::KafkaBrokerRtt) => vec![(
-                LogVariant::Materialized(MaterializedLog::SourceInfo),
-                vec![(1, 1), (2, 2), (3, 3)],
-            )],
-            LogVariant::Materialized(MaterializedLog::KafkaConsumerInfo) => vec![(
+            LogVariant::Materialized(MaterializedLog::KafkaSourceStatistics) => vec![(
                 LogVariant::Materialized(MaterializedLog::SourceInfo),
-                vec![(1, 1), (2, 2), (3, 3)],
+                vec![(1, 0)],
```
Comment (uce) [inline src/dataflow-types/src/logging.rs:245]: I'm not sure that I'm reading the code correctly, but should this be `(0, 1)`?

**Review by uce — COMMENTED** [review]
I skimmed the logging changes which LGTM.

(( Personally, I would have opted to add a short release note, but I can see that it's ok to skip since system tables are not considered stable. ))

File: src/dataflow-types/src/logging.rs (hunk 1 above)
Comment (benesch) [reply src/dataflow-types/src/logging.rs:245]: oof, yes, thank you!

File: src/coord/src/catalog/builtin.rs [hunk 2]
```diff
@@ -626,6 +610,14 @@ pub const MZ_ARRANGEMENT_RECORDS_INTERNAL: BuiltinLog = BuiltinLog {
...
 
+pub const MZ_KAFKA_SOURCE_STATISTICS: BuiltinLog = BuiltinLog {
+    name: "mz_kafka_source_statistics",
+    schema: MZ_CATALOG_SCHEMA,
+    variant: LogVariant::Materialized(MaterializedLog::KafkaSourceStatistics),
+    id: GlobalId::System(3040),
+    index_id: GlobalId::System(3041),
+};
+
 // Next id BuiltinLog: 3040
```
[the remark this answers is not in the document]
Comment (benesch) [reply src/coord/src/catalog/builtin.rs:621]: Thanks, fixed.

**Pull request merged**